core/services/gateway/connector/connector.go

```diff
@@ -125,6 +126,22 @@ func NewGatewayConnector(config *ConnectorConfig, signer Signer, handler Gateway
 	return connector, nil
 }
 
+func (c *gatewayConnector) AddHandler(methods []string, handler GatewayConnectorHandler) error {
```

Comment:
Maybe addHandlers that takes an array?

Reply:
Each capability will only add one handler (to support itself).

Reply:
Do we want to check to make sure that only 1 handler is added?